include/swift/SILOptimizer/Analysis/EpilogueARCAnalysis.h

```diff
@@ -200,9 +200,11 @@ class EpilogueARCContext {
     // We are checking for retain. If this is a self-recursion. call
     // to the function (which returns an owned value) can be treated as
     // the retain instruction.
-    if (auto *AI = dyn_cast<ApplyInst>(II))
-     if (AI->getCalleeFunction() == II->getParent()->getParent())
-       return true;
+    if (auto *AI = dyn_cast<ApplyInst>(II)) {
```

Comment:
LGTM. Can you in a follow on commit add a comment here reminding the user that arg is going to be a return value of the function since we are performing retain motion. I was confused at first b/c of getArg.